sls-versions/src/main/java/com/palantir/sls/versions/SlsVersion.java

```diff
@@ -62,8 +62,20 @@ public static boolean check(String coordinate) {
 
     public abstract int getPatchVersionNumber();
 
+    /**
+     * The first version number after major, minor, and patch. Typically either RC or snapshot version number.
```

Comment:
nits:
- Avoid abbrevations
- The "typically" seems unnecessary - we know exactly when this will be set.

```suggestion
     * The first version number after major, minor, and patch. The release candidate number, if this version is a release candidate or a release candidate snapshot, and the snapshot number, if this version is a release snapshot.
```

Reply:
> We know exactly when this will be set.

That's not quite true, since this is a field on `SlsVersion`, not on `OrderableSlsVersion`. It's not quite determined that this _must_ be empty if not an RC or a snapshot.

Reply:
These fields will always be empty on a `NonOrderableSlsVersion`, right?

https://github.com/palantir/sls-version-java/blob/bc4fc923ad7b6b6b8b620d103338eddcf708db82/sls-versions/src/main/java/com/palantir/sls/versions/NonOrderableSlsVersion.java#L48-L54

Reply:
They will, but that's not part of the definition of `SlsVersion`. This is an implementation detail. I meant that _semantically_, nothing would prevent someone from building their own `NonOrderableSlsVersion` and setting these.

With that said, that isn't done as far as a I know, and I'm not sure how you'd really even do it in the first place, considering there isn't necessarily a sequence version number for non-orderable versions.

But that's why I believe these shouldn't live in `SlsVersion` in the first place and why I'd like to move `rc` and `snapshot` back down into `OrderableSlsVersion` (since they don't make sense in `NonOrderableSlsVersion`)

Reply:
> But that's why I believe these shouldn't live in SlsVersion in the first place and why I'd like to move rc and snapshot back down into OrderableSlsVersion (since they don't make sense in NonOrderableSlsVersion)

Sure that makes sense. Putting these values in `SlsVersion` in the first place was probably a mistake.